Description: fix compiler warnings
 This patch fixes "warning: cast to pointer from integer of different size"
 in protolib/src/common/protoDispatcher.cpp by simply adding a cast
 to intptr_t as recommended by
  https://buildd.debian.org/~brlink/bytag/W-pointer-cast-size-mismatch.html.
 So far no negative effects to the workings of the program have been observed.
 More thorough tests should probably be done.
Author: Raoul Gunnar Borenius <borenius@dfn.de>
Forwarded: yes, sent private email to upstream authors
Last-Update: 2012-02-20
--- a/protolib/src/common/protoDispatcher.cpp
+++ b/protolib/src/common/protoDispatcher.cpp
@@ -760,7 +760,7 @@
     if (dp->controller) Lock(dp->controller->lock_b);
     Lock(dp->suspend_mutex);
     dp->thread_started = true;
-    ExitStatus result = (ExitStatus)dp->Run();
+    ExitStatus result = (ExitStatus)(intptr_t)dp->Run();
     Unlock(dp->suspend_mutex);
     DoThreadExit(result);
     return result;
